server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -522,19 +521,9 @@ private boolean shouldReverseLeafReaderContexts() {
         // This is actually beneficial for search queries to start search on latest segments first for time series workload.
         // That can slow down ASC order queries on timestamp workload. So to avoid that slowdown, we will reverse leaf
         // reader order here.
-        if (searchContext.indexShard().isTimeSeriesDescSortOptimizationEnabled()) {
-            // Only reverse order for asc order sort queries
-            if (searchContext.sort() != null
-                && searchContext.sort().sort != null
-                && searchContext.sort().sort.getSort() != null
-                && searchContext.sort().sort.getSort().length > 0
-                && searchContext.sort().sort.getSort()[0].getReverse() == false
-                && searchContext.sort().sort.getSort()[0].getField() != null
-                && searchContext.sort().sort.getSort()[0].getField().equals(DataStream.TIMESERIES_FIELDNAME)) {
-                return true;
-            }
-        }
-        return false;
+        return searchContext.indexShard().isTimeSeriesDescSortOptimizationEnabled()
```

Comment:
I think this 3 conditions could go under `SearchContext::shouldUseTimeSeriesDescSortOptimization`, the same you did for `SearchContext::shouldUseConcurrentSearch`

Reply:
@reta this just checks if index contains `@timestamp` field. Here we need if sort query is on @timestamp field as primary field to reverse the segment orders for only `asc` sort query. So this looks good to me.

Reply:
thanks @gashutos the  3 conditions we have should give the yes / no decision for optimization:
 - isTimeSeriesDescSortOptimizationEnabled()
 - searchContext.isSortOnTimeSeriesField()
 - searchContext.sort().sort.getSort()[0].getReverse() == false;

I suggest to extract them into a method on SearchContext::shouldUseTimeSeriesDescSortOptimization

Reply:
Ah, got it.

Reply:
Made the change in latest revision